core/libp2p/connection/connection.hpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CONNECTION_HPP
+#define KAGOME_CONNECTION_HPP
+
+#include <vector>
+
+#include <boost/optional.hpp>
+#include <rxcpp/rx-observable.hpp>
+#include "common/result.hpp"
+#include "libp2p/common_objects/multiaddress.hpp"
+#include "libp2p/common_objects/network_message.hpp"
+#include "libp2p/common_objects/peer.hpp"
+
+namespace libp2p {
+  namespace connection {
+    /**
+     * Point-to-point link to the other peer
+     */
+    class Connection {
```

Comment:
Connection and Stream share same interface (read/write), maybe connection should extend Stream? Or they both should extend from Readable / Writeable interfaces?

Reply:
Now, signatures of Read methods are different, so only Writable interface may be created

Reply:
If we'd write code in go, you'd use native go's `interface Writeable {...}`. In C++, it is up to you to create it. I am okay with or without this interface